lib/matplotlib/tests/test_axes.py

```diff
@@ -4603,6 +4603,83 @@ def test_hist_stacked_bar():
     ax.legend(loc='upper right', bbox_to_anchor=(1.0, 1.0), ncols=1)
 
 
+@pytest.mark.parametrize("histtype, color, edgecolor, facecolor,",
+                         [("step", ["b", "g", "m"], ["r", "k", "b"], ["b", "g", "m"]),
+                          ("stepfilled", ["b", "g", "m"], ["r", "k", "b"], [None]*3),
+                          ("step", ["b", "g", "m"], [None]*3, ["b", "g", "m"]),
+                          ("stepfilled", None, ["r", "k", "b"], ["b", "g", "m"]),
+                          ("stepfilled", ["b", "g", "m"], [None]*3, [None]*3),
+                          ("step", ["b", "g", "m"], [None]*3, [None]*3),
+                          ("stepfilled", None, ["r", "k", "b"], [None]*3),
+                          ("step", None, ["r", "k", "b"], [None]*3),
+                          ("stepfilled", None, [None]*3, ["b", "g", "m"]),
+                          ("step", None, [None]*3, ["b", "g", "m"])])
+@check_figures_equal(extensions=["png"])
+def test_hist_vectorized_colors(fig_test, fig_ref, histtype, color, edgecolor,
+                                facecolor):
+    np.random.seed(19680801)
+    x = [np.random.randn(n) for n in [2000, 5000, 10000]]
+
+    _, bins, _ = fig_test.subplots().hist(x, bins=10, histtype=histtype, color=color,
+                                          edgecolor=edgecolor, facecolor=facecolor)
+    ref_ax = fig_ref.subplots()
+    color = [None]*3 if color is None else color
+    edgecolor = [None]*3 if edgecolor is None else edgecolor
+    facecolor = [None]*3 if facecolor is None else facecolor
+
+    for i in range(2, -1, -1):
```

Comment:
That's hard to read. Use `reversed(range(3))` or just use exlo
```suggestion
    for i in [2, 1, 0]:
```